File: mock1-moveZeroes.js
function moveZeroes(arr) {
    var i = 0, c = 0;
    while (i < arr.length) {
        if (arr[i] != 0) {
            arr[c] = arr[i];
            c++;
        }
        i++;
    }
    while (c < arr.length) {
        arr[c++] = 0;
    }
}
var a = [ 0, 1, 0, 3, 12];
moveZeroes(a);
console.log(a);

File: mock1-movezeroes.py
def movezeroes(arr):
    c=0
    i=0
    while(i<len(arr)):
        if(arr[i]!=0):
            arr[c]=arr[i]
            c+=1
        i+=1
    while(c<len(arr)):
        arr[c]=0
        c+=1
arr=[0,1,0,3,12]
movezeroes(arr)
print(arr)
        
